src/main/java/org/jboss/remoting3/RemotingXmlParser.java

```diff
@@ -23,18 +23,25 @@
 
 import java.io.IOException;
 import java.net.URI;
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.wildfly.client.config.ClientConfiguration;
 import org.wildfly.client.config.ConfigXMLParseException;
 import org.wildfly.client.config.ConfigurationXMLStreamReader;
+import org.xnio.OptionMap;
+import org.xnio.Options;
 import org.xnio.XnioWorker;
 
 /**
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 final class RemotingXmlParser {
     private static final String NS_REMOTING_5_0 = "urn:jboss-remoting:5.0";
+    private static final String NS_REMOTING_5_1 = "urn:jboss-remoting:5.1";
```

Comment:
Please, look at EJBClientDescriptorParsingProcessor. You will see that you need to have a different parser for each namespace version, thus preventing that a ns remoting 5.0 gets parsed as if it were a 5.1.

Reply:
In this case, since it is only adding some attributes I think we can use the same parser it will complain at xsd validation it they use the new attributes, should I add a new parser anyway?

Reply:
Normally we don't bother with whole new parsers for client config (see the EJB client config parser, Elytron client config parser, XNIO, etc.).  We typically have a boolean method like `isAtLeast5_1` or whatever.

Reply:
@tmiyargi Yes, it is good practice to check in the code.
@dmlloyd is right, you can follow the code here as an example: https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/client/src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java#L1306